Classes/Views/GutterLayoutManager.swift

```diff
@@ -0,0 +1,257 @@
+//
+//  GutterLayoutManager.swift
+//  Freetime
+//
+//  Created by Weyert de Boer on 15/10/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+
+protocol GutterLayoutManagerDelegate: class {
+
+    func getLineForLine(gutterLayoutManager: GutterLayoutManager, lineNumber: Int, for line: String) -> String
+
+    func getMaximumGutterWidth(gutterLayoutManager: GutterLayoutManager) -> CGFloat
+}
+
+extension GutterLayoutManagerDelegate {
+
+    func getLineForLine(gutterLayoutManager: GutterLayoutManager, lineNumber: Int, for line: String) -> String {
+        return "\(lineNumber)"
+    }
+
+    func getMaximumGutterWidth(gutterLayoutManager: GutterLayoutManager) -> CGFloat {
+        guard let textStorage = gutterLayoutManager.textStorage else { return 0 }
+        let numberOfLines = textStorage.string.components(separatedBy: .newlines).count
+        let maxLength = "\(numberOfLines)".count
+        let fillerString = String(repeating: "9", count: maxLength)
```

Comment:
If the font is monospace, is there a better/nicer way of calculating this?

Reply:
This is the best approach I could come up. If anyone has a better idea, please don't hesitate to share!